crates/polars-arrow/src/legacy/kernels/rolling/no_nulls/mod.rs

```diff
@@ -42,27 +41,36 @@ pub(super) fn rolling_apply_agg_window<'a, Agg, T, Fo>(
 where
     Fo: Fn(Idx, WindowSize, Len) -> (Start, End),
     Agg: RollingAggWindowNoNulls<'a, T>,
-    T: Debug + NativeType,
+    T: Debug + NativeType + Num,
 {
     let len = values.len();
     let (start, end) = det_offsets_fn(0, window_size, len);
     let mut agg_window = Agg::new(values, start, end, params);
+    if let Some(validity) = create_validity(min_periods, len, window_size, &det_offsets_fn) {
+        if validity.iter().all(|x| !x) {
+            // all null!
+            return Ok(Box::new(PrimitiveArray::new(
```

Comment:
We should use `new_null` constructor here. The bitmap now first allocates a vec full of `false` and then iterates that to construct a bitmap. That can be done in a single allocation.